### DISCUSSION TITLE
Invalid ELF header with serverless-python-requirements, docker, bcrypt

### DISCUSSION THREAD

**Discussion by oddanderson** [serverless/serverless#10434] [Q&A]
I followed the guide here: https://www.serverless.com/blog/serverless-python-packaging/

which means I am using serverless to deploy my python-based lambda functions. I am using Docker (via the guide) to package requirements up. However, I am running into the following error after I deploy and hit a function:

`[ERROR] Runtime.ImportModuleError: Unable to import module 'auth': /var/task/bcrypt/_bcrypt.abi3.so: invalid ELF header
Traceback (most recent call last):`

bcrypt is referenced in my python code using
`import bcrypt` 
and in my requirements.txt using 
```
bcrypt==3.2.0
cffi==1.15.0
pycparser==2.21
PyMySQL==1.0.2
six==1.16.0
```

here is my yml file for reference
```
plugins:
  - serverless-python-requirements
custom:
  pythonRequirements:
    dockerizePip: non-linux
 ```

I've tried a dozen or so guides on how to fix this. I've tried building the venv directory from an EC2 machine, adding node_modules to .dockerignore, but I'm pretty stuck. I'm at the point where I'm about to ditch serverless because I think I can solve this issue if I package and zip everything myself.

**Comment by pgrzesik** [comment]
Hello @oddanderson - could you share a bit more on the environment in which you're building your application (OS, output of `serverless --version`)?
